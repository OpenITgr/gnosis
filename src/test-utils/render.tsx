/* eslint-disable import/export */
import React, { FC, ReactElement } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
