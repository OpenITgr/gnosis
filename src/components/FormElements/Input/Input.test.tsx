import React from "react";
import userEvent from "@testing-library/user-event";
import { Default as Input, WithIconBefore, WithIconAfter } from "./Input.stories";
import { screen, render } from "@test-utils/render";

describe("<Input />", () => {
  it("renders correctly", () => {
    render(<Input {...Input.args} />);

    const label = screen.getByText("Username");
    const input = screen.getByLabelText("Username");

    expect(label).toHaveTextContent("Username");
    expect(input).toHaveAttribute("placeholder", "Your LMS username");
  });

  it("changes the input's value", () => {
    render(<Input {...Input.args} />);
    const input = screen.getByLabelText("Username");

    userEvent.type(input, "new value");

    expect(input).toHaveValue("new value");

    userEvent.clear(input);

    expect(input).toHaveValue("");
  });

  it("renders with icon before", () => {
    render(<WithIconBefore {...WithIconBefore.args} />);

    const iconBefore = screen.getByTestId(/input-icon-before/i);

    expect(iconBefore).toBeInTheDocument();
  });

  it("renders with icon after", () => {
    render(<WithIconAfter {...WithIconAfter.args} />);

    const iconAfter = screen.getByTestId(/input-icon-after/i);

    expect(iconAfter).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });
});
