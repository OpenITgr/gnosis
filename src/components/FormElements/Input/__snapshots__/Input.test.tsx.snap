// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<div>
  <div
    class="valid css-38lqwp-inputContainerBaseStyles-inputContainerBaseStyles-inputBaseStyles-inputBaseStyles-inputContainer-inputContainer"
  >
    <div
      class="input-wrapper"
    >
      <input
        id="story-input"
      />
    </div>
  </div>
</div>
`;
